File: src/hello.ml
[@@@bs.config {jsx = 3}]
module Counter =
  struct
    let make ~name  =
      let (count,setCount) = React.useState (fun ()  -> 0) in
      ((div
          ~children:[((p
                         ~children:[React.string
                                      (name ^
                                         (((" clicked ")[@reason.raw_literal
                                                          " clicked "])
                                            ^
                                            ((string_of_int count) ^
                                               ((" times")[@reason.raw_literal
                                                            " times"]))))] ())
                    [@JSX ]);
                    ((button
                        ~onClick:(fun _  -> setCount (fun _  -> count + 1))
                        ~children:[React.string
                                     (("Click me")[@reason.raw_literal
                                                    "Click me"])] ())
                    [@JSX ])] ())[@JSX ])[@@react.component ]
  end
let _ =
  ReactDOMRe.renderToElementWithId
    ((Counter.createElement
        ~name:(("Counter")[@reason.raw_literal "Counter"]) ~children:[] ())
    [@JSX ]) (("preview")[@reason.raw_literal "preview"])

File: src/hello.js
'use strict';

var React = require("react");
var ReactDOMRe = require("reason-react/src/legacy/ReactDOMRe.bs.js");
var ReasonReact = require("reason-react/src/legacy/ReasonReact.bs.js");

var component = ReasonReact.statelessComponent("Greeting");

function make(_children) {
  return {
    debugName: component.debugName,
    reactClassInternal: component.reactClassInternal,
    handedOffState: component.handedOffState,
    willReceiveProps: component.willReceiveProps,
    didMount: component.didMount,
    didUpdate: component.didUpdate,
    willUnmount: component.willUnmount,
    willUpdate: component.willUpdate,
    shouldUpdate: component.shouldUpdate,
    render: (function (_self) {
      return React.createElement("p", undefined, "Hello");
    }),
    initialState: component.initialState,
    retainedProps: component.retainedProps,
    reducer: component.reducer,
    jsElementWrapped: component.jsElementWrapped
  };
}

ReactDOMRe.renderToElementWithId(ReasonReact.element(undefined, undefined, make([])), "root");

exports.component = component;
exports.make = make;
/* component Not a pure module */
